### ISSUE TITLE
Deleting a task does not work

### ISSUE THREAD

**Issue by michael-lurquin** [laravel/quickstart-intermediate#12]
In the "Policies", the "destroy" method contains this:

```
return $user->id === $task->user_id;
```

Out, "$task->user_id" is a string that is compared to "$user->id" which is an integer.

It should therefore replace the line:

```
return $user->id === (int)$task->user_id;
```

or

```
return $user->id == $task->user_id;
```

**Comment by halyc0n** [comment]
I had the same problem

**Comment by LudoVio** [comment]
Same problem for me, I checked in the database, and task.user_id is int(11), so it's weird that we get a string instead of an int in php.

**Comment by LudoVio** [comment]
Just read that MySql return all values as string on query, but sometimes (on creating element and probably with the automagicaly passed $user arg of policy's method) laravel use special value returned by the database driver like "lastInsertId" or "insertGetId" and cast it to int.

"Since Laravel 5 you can add the $casts property to your model, to cast its attributes properly. In this case it should be something like this:"

```php
protected $casts = [ 'id' => 'integer' ];
```

source: https://laracasts.com/discuss/channels/general-discussion/modelcreate-returns-id-as-integer-while-modelfind-return-it-as-string

**Comment by petermichaux** [comment]
I used MySQL for the tutorial and used `===` as written in the tutorial. I didn't have any problems.

**Comment by anandhrushikesh** [comment]
@petermichaux what is your MySQL version? 
I am running MySQL 5.5.48 and have this error.

**Comment by petermichaux** [comment]
```
$ mysql --version
mysql  Ver 14.14 Distrib 5.7.10, for osx10.9 (x86_64) using  EditLine wrapper
```

**Comment by anandhrushikesh** [comment]
Upgraded to 5.6.29 issue persists.
Again upgraded to 5.7.11, issue persists. 

I'm running Percona:
Server version:         5.7.11-4 Percona Server (GPL), Release '4', Revision '5c940e1'

**Comment by anandhrushikesh** [comment]
Uninstalled Percona 5.7.11-4.  
Installed MySQL Community Server 5.7.12 and the issue persists.

Any pointers on where to debug next?

**Comment by anandhrushikesh** [comment]
New code has `user_id` cast as 'int':
https://github.com/laravel/quickstart-intermediate/blob/master/app/Task.php

`protected $casts = [
        'user_id' => 'int',
    ];
`

**Comment by ykws** [comment]
@michael-lurquin Resolved #26
